### DISCUSSION TITLE
how to achieve smooth movement of a sprite with keyboard control?

### DISCUSSION THREAD

**Discussion by 10100010010001001010** [axmolengine/axmol#1822] [Q&A]
I needed to make keyboard controls, but I discovered that simply moving the sprite by some value is essentially teleportation and there is no smooth movement. What would you do in this case?

So far I've implemented it like this:
```
void MainScene::onKeyPressed(EventKeyboard::KeyCode code, Event* event)
{
    auto sprite = getChildByName<spine::SkeletonAnimation*>("sprite");
    auto pos    = sprite->getPosition();

    switch (code)
    {
    case EventKeyboard::KeyCode::KEY_UP_ARROW:
        pos += Vec2(0, 10.0f);
        break;

    case EventKeyboard::KeyCode::KEY_LEFT_ARROW:
        pos += Vec2(-10.0f, 0);
        break;

    case EventKeyboard::KeyCode::KEY_DOWN_ARROW:
        pos += Vec2(0, -10.0f);
        break;

    case EventKeyboard::KeyCode::KEY_RIGHT_ARROW:
        pos += Vec2(10.0f, 0);
        break;
    }

    auto moveAction = MoveTo::create(1.0f, pos); 
    sprite->runAction(moveAction);
}
```

**Comment by rh101** [comment]
Try stopping the previous move action before running a new one.  The issue may be that onKeyPressed is constantly called, so you're just adding more and more actions to the same node.  

What you actually want is to simply move it from its current position by an amount to the new position, so you need to cancel the current action on the node before applying the new action.
